src/persistedState/persistedStateService.ts

```diff
@@ -13,21 +14,23 @@ export class PersistedStateService {
 		private readonly getStorageUri: () => Uri | null,
 		private readonly jobManager: JobManager,
 		private readonly messageBus: MessageBus,
+		private readonly __repositoryService: RepositoryService,
 	) {
 		const debouncedOnUpdateElementsMessage = debounce(
-			() => this.#onUpdateElementsMessage(),
+			() => this.saveExtensionState(),
 			1000,
 		);
 
 		this.messageBus.subscribe(MessageKind.updateElements, () =>
 			debouncedOnUpdateElementsMessage(),
 		);
+
 		this.messageBus.subscribe(MessageKind.clearState, () =>
 			this.#onClearStateMessage(),
 		);
 	}
 
-	async #onUpdateElementsMessage() {
+	public async saveExtensionState() {
```

Comment:
why did you change the method name? actually it made sense to have it connected to the message kind

Reply:
I wanted to call this method directly so i made it public and changed the name to neutral. Should i add new events instead, and subscribe to them instead?